sympy/geometry/ellipse.py

```diff
@@ -147,6 +147,15 @@ def __new__(
         if hradius == vradius:
             return Circle(center, hradius, **kwargs)
 
+        if hradius == 0 and vradius == 0:
+            return center
+
+        if hradius == 0:
+            return Segment(Point(center[0], center[1]-vradius), Point(center[0], center[1]+vradius))
+
+        if vradius == 0:
+            return Segment(Point(center[0]-hradius, center[1]), Point(center[0]+hradius, center[1]))
```

Comment:
How about:
```
if hradius == vradius:
	return center if hradius == 0 else Circle(center, hradius, **kwargs)

if hradius == 0 or vradius == 0:
	return Segment(Point(center[0] - hradius, center[1] - vradius),
						Point(center[0] + hradius, center[1] + vradius))
```

Reply:
You could possibly use `length` for the test case where a `Segment` is returned by a degenerate `Ellipse`.